### ISSUE TITLE
Default Loss Function

### ISSUE THREAD

**Issue by DoctorSlimm** [UKPLab/sentence-transformers#2495]
Hiiii was wondering what the default loss function is? + how to implement custom in the framework?

ps: i know this is like totally random so apologies

Notably have been reading the [Atlas paper on fine tuning retrievers] (https://jmlr.org/papers/volume24/23-0037/23-0037.pdf) and the LOOL loss function is interesting:

<img width="647" alt="Screenshot 2024-02-21 at 08 49 42" src="https://github.com/UKPLab/sentence-transformers/assets/87606888/e2afeb0f-5e2b-42f9-8852-fa03a8b0229b">
<img width="628" alt="Screenshot 2024-02-21 at 08 50 02" src="https://github.com/UKPLab/sentence-transformers/assets/87606888/114c6db8-d5c4-44cb-a7fc-310eea01fca6">

**Comment by tomaarsen** [comment]
Hello!

Sentence Transformers doesn't have a "default" loss function per se, users always have to specify one. The current options are listed right [here](https://sbert.net/docs/package_reference/losses.html). They're all stored over here: https://github.com/UKPLab/sentence-transformers/tree/master/sentence_transformers/losses

Implementing a custom loss involves creating a class that:
* is a subclass of `torch.nn.Module`,
* accepts a `model` parameter as the first option (this is the Sentence Transformer model) & optionally more parameters afterwards,
* implements a `forward` method that has the `self, sentence_features: Iterable[Dict[str, Tensor]], labels: Tensor` signature. The `sentence_features` is a list that has a dictionary of whatever the model requires (usually "input_ids" and "attention_mask"). You can look at other losses for examples of how this is used to get the embeddings.
* the `forward` method must return a loss value. It must be a `Tensor` so that the model can learn using backpropagation.

---

- Tom Aarsen
